virtual_rainforest/core/logger.py

```diff
@@ -76,10 +76,45 @@
 """  # noqa: D205, D415
 
 import logging
+from pathlib import Path
 
 logging.basicConfig(
     level=logging.INFO,
     format="[%(levelname)s] - %(module)s - %(funcName)s(%(lineno)d) - %(message)s",
 )
 
 LOGGER = logging.getLogger("virtual_rainforest")
+
+
+def add_file_logger(logfile: Path) -> None:
+    """Redirect logging to a provided file path.
+
+    Args:
+      logfile: The path to a file to use for logging.
+    """
+
+    # Do not propogate errors up to parent handler - this avoids mirroring the log
+    # output through the StreamHandler associated with the root logger
+    LOGGER.propagate = False
+
+    # Add a specific file handler for this log.
+    format = "[%(levelname)s] - %(module)s - %(funcName)s(%(lineno)d) - %(message)s"
+    formatter = logging.Formatter(fmt=format)
+    handler = logging.FileHandler(logfile)
+    handler.setFormatter(formatter)
+    handler.name = "vr_logfile"
+    LOGGER.addHandler(handler)
+    LOGGER.setLevel(logging.DEBUG)
+
+
+def remove_file_logger() -> None:
+    """Remove an existing file logger and return to stream logging."""
+
+    # Find the file logger by name and remove it
+    vr_logfile = [
+        handler for handler in LOGGER.handlers if handler.name == "vr_logfile"
+    ]
+    LOGGER.removeHandler(vr_logfile[0])
```

Comment:
You don't need to make a list here:

```suggestion
    vr_logfile = next(
        handler for handler in LOGGER.handlers if handler.name == "vr_logfile"
    )
    LOGGER.removeHandler(vr_logfile)
```